sys/sys_unix.go

```diff
@@ -0,0 +1,8 @@
+// +build linux darwin
+
+package sys
+
+const (
+	Root    = "/"
+	ExecExt = ""
+)
```

Comment:
I don't think these two constants deserve their own package. Can we move them to `cmd`? If there are import issues I think it is okay to repeat these constants in other packages. I would prioritize keeping the import graph tidy. There may be cases like the usage in `restorer` where we could remove them entirely (we can make `UntarLayer` default to the root when given an empty destination).

Reply:
The benefit to having these in their own package is that it addresses two concerns:
- Avoids import cycles since this can be seen as a "leaf" node in the import tree
- This was created when addressing the concern about keeping the default paths in sync between OSes (refactored separate OS-dependent files to a single file that uses `filepath.Join`, etc. in combination with these constants)

It seems the second bullet would sort of go against repeating these constants in multiple places. I'll see if I can move them to a different package.

Reply:
So tinkering a bit, `cmd` seems like too high-level of a package to move this to without causing import cycles. How important is this? It feels like one of those cases that can go either way, depending on stylistic opinions. I'd argue that the import tree is simpler to understand and use with this as a leaf node (i.e. modular, self-contained packages), but other folks might argue that simpler means fewer packages altogether.

I see a few options:
- Leave as is
- Find a lower-level package where this could live instead
- Force it into command and spend time debugging import cycles

Thoughts?

Reply:
@ekcasey Actually, it worked pretty nicely to move into the `env` package. That feels like a reasonable compromise. Let me know what you think :)

Reply:
> It seems the second bullet would sort of go against repeating these constants in multiple places. I'll see if I can move them to a different package though.

I feel like this concern is less relevant for these constants than for the default input values. We will realistically change default input values at some point and we don't want linux/windows values to fall out of sync. The posix and windows fs root dirs on the other hand, should not change. This is how I would handle it:

https://github.com/buildpacks/lifecycle/commit/035a3c0909ef32ce9d63983573203d048b70a5f7

I have removed some unnecessary usage of these constants leaving us with the original motivating usage in `cmd` and one other usage in `archive` (since we already have very specific OS-dependent path logic there it doesn't feel crazy to me to redefine these in that context). Then

Reply:
My main concern is that when these are exported they become part of the public package level interface for `env` and I don't think we want that. I would like the keep the import graph and the godoc as clean and straightforward as possible.

Reply:
OK. Makes sense. We'll pull in your suggested change :)